scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt

```diff
@@ -122,13 +112,30 @@ class ClearlyDefinedStorage(
                 coordinates.revision.orEmpty()
             )
 
-            val version = findScanCodeVersion(tools, coordinates)
-            if (version != null) {
-                listOfNotNull(loadScanCodeResult(coordinates, version))
-            } else {
-                logger.debug { "$coordinates was not scanned with any version of ScanCode." }
+            val toolsVersionsByName = tools.mapNotNull { it.withoutPrefix("$coordinates/") }
+                .groupBy({ it.substringBefore('/') }, { it.substringAfter('/') })
+                .mapValues { (_, versions) -> versions.sortedWith(AlphaNumericComparator) }
+
+            toolsVersionsByName.mapNotNull { (name, versions) ->
+                // See https://github.com/clearlydefined/service#tool-name-registry.
+                when (name) {
+                    "scancode" -> {
+                        loadToolData(coordinates, name, versions.last())?.let { result ->
+                            val summary = generateSummary(SpdxConstants.NONE, result)
+                            val details = generateScannerDetails(result)
+
+                            val provenance = getProvenance(coordinates)
+
+                            ScanResult(provenance, details, summary)
+                        }
```

Comment:
Is the case when `loadToolData()` returns null unexpected and if so should it be logged?